### PR TITLE
Move from setup.py to pyproject.toml

### PR DESCRIPTION
**Pull request by BertR**
Because of how pip and setuptools deal with git dependencies,
we sometimes ran into issues (`pip install .` vs `python setup.py install` would yield different results.)

By moving to pyproject.toml, we can have a single reference to the nbgrader commit we want to install.
`pip install .` will reference the dependencies in pyproject.toml
`python setup.py install` will no longer work.

### CODE REVIEW

**Comment by perllaghu** [comment]
There's a merge conflict still sitting in pages.py

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+[build-system]
+requires = ["flit_core >=3.2,<4"]
+build-backend = "flit_core.buildapi"
+
+
+[project]
+name = "nbexchange"
+version = "1.2.0"
```
Comment (lzach) [inline pyproject.toml:8]: Should this be 1.2.1 before merging?

**Review by lzach — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+description = "Jupyterhub extension that provides an exchange service for nbgrader."
```
Comment (perllaghu) [inline pyproject.toml:9]: Is this description still valid? It's no longer a 'hub extension.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+readme = "README.md"
+requires-python = ">=3.7"
+license = {file = "LICENSE"}
+keywords = ["nbgrader", "jupyter", "notebook"]
+authors = [
```
Comment (perllaghu) [inline pyproject.toml:14]: Am I really the author?
Normally we list "edina" :)

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  {email = "Ian.Stuart@ed.ac.uk"},
+  {name = "Ian Stuart"}
+]
+maintainers = [
```
Comment (perllaghu) [inline pyproject.toml:18]: Plus Andrew

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  "pytest-tornado",
+  "pytest-docker-tools",
+  "beautifulsoup4",
+  "html5lib",
+  "psycopg2-binary",
+  "mock",
+]
+
+[project.urls]
+homepage = "https://edina.ac.uk"
```
Comment (perllaghu) [inline pyproject.toml:58]: in packages.json, this was github

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+documentation = "https://nbexchange.readthedocs.org"
```
Comment (perllaghu) [inline pyproject.toml:59]: can you create an `index.rst` or `README.rst` file in `/docs` - referencing the other two files - as this is what _readthedocs_ needs

**Pull request merged**